### PR TITLE
Use custom bucket size for indexEntriesPerChunk

### PR DESCRIPTION
**Pull request by jaybatra26**
Currently, the variable registers a new Histogram using
default bucket sizes that have floating size as well {.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10
Since this variable measures quantities which is always a whole number,
the default buckets go unused and are not necessarily required.
This PR uses custom bucket for variable indexEntriesPerChunk with
sizes {1, 2, 4, 8, 16}

Signed-off-by: Jay Batra <jaybatra73@gmail.com>

**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
   * `cortex_ingester_received_files`
   * `cortex_ingester_received_bytes_total`
   * `cortex_ingester_sent_bytes_total`
+* [CHANGE] Use custom histogram Bucket for variable in series store`indexLookupsPerQuery`#3021
```
Comment (pracucci) [inline CHANGELOG.md:10]: When you write a CHANGELOG entry try to always express the change as understandeable from a user who has never seen Cortex internals. In this case it's better to mention the metric name.

```suggestion
* [CHANGE] The buckets for the `cortex_chunk_store_index_lookups_per_query` have been changed to 1, 2, 4, 8, 16. #3021
```

File: pkg/chunk/series_store.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ var (
 		Namespace: "cortex",
 		Name:      "chunk_store_index_lookups_per_query",
 		Help:      "Distribution of #index lookups per query.",
-		Buckets:   prometheus.DefBuckets,
+		Buckets:   prometheus.ExponentialBuckets(1, 2, 5),
```
Comment (pracucci) [inline pkg/chunk/series_store.go:39]: Note to myself. Default buckets are:
`.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10`

File: CHANGELOG.md (hunk 1 above)
Comment (jaybatra26) [reply CHANGELOG.md:10]: Thanks @pracucci . I will update this. :)

**Comment by jaybatra26** [comment]
@pracucci and @bboreham Done.
Please take a look.

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,7 @@
   * `cortex_ingester_received_files`
   * `cortex_ingester_received_bytes_total`
   * `cortex_ingester_sent_bytes_total`
+* [CHANGE] The buckets for the `cortex_chunk_store_index_lookups_per_query` have been changed to 1, 2, 4, 8, 16. #3021
```
Comment (pracucci) [inline CHANGELOG.md:10]:
```suggestion
* [CHANGE] The buckets for the `cortex_chunk_store_index_lookups_per_query` metric have been changed to 1, 2, 4, 8, 16. #3021
```

**Review by pstibrany — APPROVED** [review]

**Pull request merged**